cmake/std/atdm/test/unit_tests/set_build_options_unit_tests.sh

```diff
@@ -0,0 +1,484 @@
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FINISHED_SET_BUILD_OPTIONS} 1
+}
+
+
+testCompilerClang() {
+
+  ATDM_CONFIG_BUILD_NAME=clang-5.0.1
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CLANG-5.0.1
+
+  ATDM_CONFIG_BUILD_NAME=before_clang-5.0.1_after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CLANG-5.0.1
+
+  ATDM_CONFIG_BUILD_NAME=before-clang-5.0.1-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CLANG-5.0.1
+
+  ATDM_CONFIG_BUILD_NAME=clang-7.0.1
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CLANG-7.0.1
+
+}
+
+
+testCompilerCuda() {
+
+  ATDM_CONFIG_BUILD_NAME=cuda
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CUDA
+
+  ATDM_CONFIG_BUILD_NAME=cuda-9.2
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CUDA-9.2
+
+  ATDM_CONFIG_BUILD_NAME=cuda-9.2-gnu-7.2.0
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CUDA-9.2_GNU-7.2.0
+
+  ATDM_CONFIG_BUILD_NAME=cuda-9.2_gnu-7.2.0
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CUDA-9.2_GNU-7.2.0
+
+  ATDM_CONFIG_BUILD_NAME=cuda-10.0-gnu-7.4.0
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} CUDA-10.0_GNU-7.4.0
+
+}
+
+
+testCompilerIntel() {
+
+  ATDM_CONFIG_BUILD_NAME=intel
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} INTEL
+
+  ATDM_CONFIG_BUILD_NAME=intel-17
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} INTEL-17.0.1
+
+  ATDM_CONFIG_BUILD_NAME=intel-17.0.1
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} INTEL-17.0.1
+
+  ATDM_CONFIG_BUILD_NAME=intel-18
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} INTEL-18.0.5
+
+}
+
+
+testKokkosArch() {
+
+  # Test first arch and defaults for everything else
+  ATDM_CONFIG_BUILD_NAME=default-AMDAVX
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPILER} DEFAULT
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_KOKKOS_ARCH} AMDAVX
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_BUILD_TYPE} DEBUG
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_NODE_TYPE} SERIAL
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_OPENMP} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_CUDA} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_USE_PTHREADS} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
```

Comment:
Is this a duplicate of line 134?

Reply:
Yes, you are correct!  I will remove the extra one.